app/components/wallet/AccountsOverview.tsx

```diff
@@ -141,6 +187,42 @@ const AccountsOverview = () => {
   const balance = balances[address];
   const { value, unit } = parseSmidge(balance?.currentState?.balance || 0);
 
+  const handleCopy = async (e: React.MouseEvent, value: string) => {
+    e.stopPropagation();
+    await navigator.clipboard.writeText(value);
+    setIsCopied(value);
+    clearTimeout(copyTimeout);
+    copyTimeout = setTimeout(() => setIsCopied(null), 3000);
+  };
```

Comment:
Let's avoid code duplication and extract all these code in separate components `basicComponents/copyButton.tsx` and `basicComponents/explorerButton.tsx`.
They also should be reused in `Address` component.